Test errors from a new installation

I've just installed boulder following the instructions in the README and here's the failing tests:

```
➜  boulder git:(master) ✗ ./test.sh
[             vet] Starting
go vet github.com/letsencrypt/boulder/akamai github.com/letsencrypt/boulder/bdns github.com/letsencrypt/boulder/ca github.com/letsencrypt/boulder/cdr github.com/letsencrypt/boulder/cmd github.com/letsencrypt/boulder/cmd/admin-revoker github.com/letsencrypt/boulder/cmd/boulder-ca github.com/letsencrypt/boulder/cmd/boulder-publisher github.com/letsencrypt/boulder/cmd/boulder-ra github.com/letsencrypt/boulder/cmd/boulder-sa github.com/letsencrypt/boulder/cmd/boulder-va github.com/letsencrypt/boulder/cmd/boulder-wfe github.com/letsencrypt/boulder/cmd/caa-checker github.com/letsencrypt/boulder/cmd/caa-checker/proto github.com/letsencrypt/boulder/cmd/caa-checker/test-client github.com/letsencrypt/boulder/cmd/cert-checker github.com/letsencrypt/boulder/cmd/contact-exporter github.com/letsencrypt/boulder/cmd/expiration-mailer github.com/letsencrypt/boulder/cmd/expired-authz-purger github.com/letsencrypt/boulder/cmd/notify-mailer github.com/letsencrypt/boulder/cmd/ocsp-responder github.com/letsencrypt/boulder/cmd/ocsp-updater github.com/letsencrypt/boulder/cmd/orphan-finder github.com/letsencrypt/boulder/cmd/rabbitmq-setup github.com/letsencrypt/boulder/cmd/single-ocsp github.com/letsencrypt/boulder/cmd/single-ocsp/test github.com/letsencrypt/boulder/core github.com/letsencrypt/boulder/core/proto github.com/letsencrypt/boulder/csr github.com/letsencrypt/boulder/goodkey github.com/letsencrypt/boulder/grpc github.com/letsencrypt/boulder/grpc/creds github.com/letsencrypt/boulder/log github.com/letsencrypt/boulder/mail github.com/letsencrypt/boulder/metrics github.com/letsencrypt/boulder/mocks github.com/letsencrypt/boulder/nonce github.com/letsencrypt/boulder/policy github.com/letsencrypt/boulder/probs github.com/letsencrypt/boulder/publisher github.com/letsencrypt/boulder/publisher/proto github.com/letsencrypt/boulder/ra github.com/letsencrypt/boulder/ratelimit github.com/letsencrypt/boulder/reloader github.com/letsencrypt/boulder/rpc github.com/letsencrypt/boulder/sa github.com/letsencrypt/boulder/sa/satest github.com/letsencrypt/boulder/test github.com/letsencrypt/boulder/test/ct-test-srv github.com/letsencrypt/boulder/test/dns-test-srv github.com/letsencrypt/boulder/test/mail-test-srv github.com/letsencrypt/boulder/test/vars github.com/letsencrypt/boulder/va github.com/letsencrypt/boulder/va/proto github.com/letsencrypt/boulder/wfe
[             vet] Done
[             fmt] Starting
check_gofmt
[             fmt] Done
[      migrations] Starting
./test/test-no-outdated-migrations.sh
[      migrations] Done
go test -p 1 github.com/letsencrypt/boulder/akamai github.com/letsencrypt/boulder/bdns github.com/letsencrypt/boulder/ca github.com/letsencrypt/boulder/cdr github.com/letsencrypt/boulder/cmd github.com/letsencrypt/boulder/cmd/admin-revoker github.com/letsencrypt/boulder/cmd/boulder-ca github.com/letsencrypt/boulder/cmd/boulder-publisher github.com/letsencrypt/boulder/cmd/boulder-ra github.com/letsencrypt/boulder/cmd/boulder-sa github.com/letsencrypt/boulder/cmd/boulder-va github.com/letsencrypt/boulder/cmd/boulder-wfe github.com/letsencrypt/boulder/cmd/caa-checker github.com/letsencrypt/boulder/cmd/caa-checker/proto github.com/letsencrypt/boulder/cmd/caa-checker/test-client github.com/letsencrypt/boulder/cmd/cert-checker github.com/letsencrypt/boulder/cmd/contact-exporter github.com/letsencrypt/boulder/cmd/expiration-mailer github.com/letsencrypt/boulder/cmd/expired-authz-purger github.com/letsencrypt/boulder/cmd/notify-mailer github.com/letsencrypt/boulder/cmd/ocsp-responder github.com/letsencrypt/boulder/cmd/ocsp-updater github.com/letsencrypt/boulder/cmd/orphan-finder github.com/letsencrypt/boulder/cmd/rabbitmq-setup github.com/letsencrypt/boulder/cmd/single-ocsp github.com/letsencrypt/boulder/cmd/single-ocsp/test github.com/letsencrypt/boulder/core github.com/letsencrypt/boulder/core/proto github.com/letsencrypt/boulder/csr github.com/letsencrypt/boulder/goodkey github.com/letsencrypt/boulder/grpc github.com/letsencrypt/boulder/grpc/creds github.com/letsencrypt/boulder/log github.com/letsencrypt/boulder/mail github.com/letsencrypt/boulder/metrics github.com/letsencrypt/boulder/mocks github.com/letsencrypt/boulder/nonce github.com/letsencrypt/boulder/policy github.com/letsencrypt/boulder/probs github.com/letsencrypt/boulder/publisher github.com/letsencrypt/boulder/publisher/proto github.com/letsencrypt/boulder/ra github.com/letsencrypt/boulder/ratelimit github.com/letsencrypt/boulder/reloader github.com/letsencrypt/boulder/rpc github.com/letsencrypt/boulder/sa github.com/letsencrypt/boulder/sa/satest github.com/letsencrypt/boulder/test github.com/letsencrypt/boulder/test/ct-test-srv github.com/letsencrypt/boulder/test/dns-test-srv github.com/letsencrypt/boulder/test/mail-test-srv github.com/letsencrypt/boulder/test/vars github.com/letsencrypt/boulder/va github.com/letsencrypt/boulder/va/proto github.com/letsencrypt/boulder/wfe
ok  	github.com/letsencrypt/boulder/akamai	0.020s
ok  	github.com/letsencrypt/boulder/bdns	0.228s
ok  	github.com/letsencrypt/boulder/ca	0.130s
ok  	github.com/letsencrypt/boulder/cdr	0.036s
ok  	github.com/letsencrypt/boulder/cmd	0.023s
ok  	github.com/letsencrypt/boulder/cmd/admin-revoker	0.022s
ok  	github.com/letsencrypt/boulder/cmd/boulder-ca	0.021s
ok  	github.com/letsencrypt/boulder/cmd/boulder-publisher	0.027s
ok  	github.com/letsencrypt/boulder/cmd/boulder-ra	0.024s
ok  	github.com/letsencrypt/boulder/cmd/boulder-sa	0.022s
ok  	github.com/letsencrypt/boulder/cmd/boulder-va	0.023s
ok  	github.com/letsencrypt/boulder/cmd/boulder-wfe	0.020s
ok  	github.com/letsencrypt/boulder/cmd/caa-checker	0.021s
?   	github.com/letsencrypt/boulder/cmd/caa-checker/proto	[no test files]
?   	github.com/letsencrypt/boulder/cmd/caa-checker/test-client	[no test files]
I110436 cert-checker.test loading hostname policy, sha256: 3ddc932620e3f59623af52971fcc3cf38c9759f64575fd018a7097875427ea5e
	main_test.go:207: Couldn't add certificate: Error 1142: INSERT command denied to user 'sa'@'localhost' for table 'fqdnsets'
--- FAIL: TestGetAndProcessCerts (0.36s)
{
  "good-certs": 2,
  "bad-certs": 1,
  "entries": {
    "020000000000004b475da49b91da5c17": {
      "valid": true
    },
    "020000000000004d1613e581432cba7e": {
      "valid": true
    },
    "020000000000004e402bc21035c6634a": {
      "valid": false,
      "problems": [
        "None really..."
      ]
    }
  }
}
FAIL
FAIL	github.com/letsencrypt/boulder/cmd/cert-checker	0.536s
ok  	github.com/letsencrypt/boulder/cmd/contact-exporter	0.181s
	main_test.go:357: Couldn't add fqdnStatusD: Error 1146: Table 'boulder_sa_test.fqdnsets' doesn't exist
--- FAIL: TestFindExpiringCertificates (0.12s)
	main_test.go:506: Couldn't add fqdnStatus 000000000000000000000000000000000538: Error 1146: Table 'boulder_sa_test.fqdnsets' doesn't exist
--- FAIL: TestCertIsRenewed (0.07s)
FAIL
FAIL	github.com/letsencrypt/boulder/cmd/expiration-mailer	0.532s
ok  	github.com/letsencrypt/boulder/cmd/expired-authz-purger	0.115s
ok  	github.com/letsencrypt/boulder/cmd/notify-mailer	0.033s
ok  	github.com/letsencrypt/boulder/cmd/ocsp-responder	0.034s
	main_test.go:107: Couldn't add test-cert.pem: Error 1142: INSERT command denied to user 'sa'@'localhost' for table 'fqdnsets'
--- FAIL: TestGenerateAndStoreOCSPResponse (0.06s)
	main_test.go:135: Couldn't add test-cert.pem: Error 1142: INSERT command denied to user 'sa'@'localhost' for table 'fqdnsets'
--- FAIL: TestGenerateOCSPResponses (0.06s)
	main_test.go:162: Couldn't add test-cert.pem: Error 1142: INSERT command denied to user 'sa'@'localhost' for table 'fqdnsets'
--- FAIL: TestFindStaleOCSPResponses (0.06s)
	main_test.go:190: Couldn't add test-cert.pem: Error 1142: INSERT command denied to user 'sa'@'localhost' for table 'fqdnsets'
--- FAIL: TestGetCertificatesWithMissingResponses (0.06s)
	main_test.go:205: Couldn't add test-cert.pem: Error 1142: INSERT command denied to user 'sa'@'localhost' for table 'fqdnsets'
--- FAIL: TestFindRevokedCertificatesToUpdate (0.07s)
	main_test.go:227: Couldn't add test-cert.pem: Error 1142: INSERT command denied to user 'sa'@'localhost' for table 'fqdnsets'
--- FAIL: TestNewCertificateTick (0.07s)
	main_test.go:246: Couldn't add test-cert.pem: Error 1142: INSERT command denied to user 'sa'@'localhost' for table 'fqdnsets'
--- FAIL: TestOldOCSPResponsesTick (0.05s)
	main_test.go:266: Couldn't add test-cert.pem: Error 1142: INSERT command denied to user 'sa'@'localhost' for table 'fqdnsets'
--- FAIL: TestMissingReceiptsTick (0.05s)
	main_test.go:346: Couldn't add test-cert.pem: Error 1142: INSERT command denied to user 'sa'@'localhost' for table 'fqdnsets'
--- FAIL: TestRevokedCertificatesTick (0.05s)
	main_test.go:372: Couldn't add test-cert.pem: Error 1142: INSERT command denied to user 'sa'@'localhost' for table 'fqdnsets'
--- FAIL: TestStoreResponseGuard (0.07s)
FAIL
FAIL	github.com/letsencrypt/boulder/cmd/ocsp-updater	0.689s
ok  	github.com/letsencrypt/boulder/cmd/orphan-finder	0.027s
?   	github.com/letsencrypt/boulder/cmd/rabbitmq-setup	[no test files]
ok  	github.com/letsencrypt/boulder/cmd/single-ocsp	0.020s
?   	github.com/letsencrypt/boulder/cmd/single-ocsp/test	[no test files]
ok  	github.com/letsencrypt/boulder/core	4.645s
?   	github.com/letsencrypt/boulder/core/proto	[no test files]
ok  	github.com/letsencrypt/boulder/csr	0.132s
ok  	github.com/letsencrypt/boulder/goodkey	10.000s
ok  	github.com/letsencrypt/boulder/grpc	0.528s
ok  	github.com/letsencrypt/boulder/grpc/creds	0.083s
ok  	github.com/letsencrypt/boulder/log	0.117s
ok  	github.com/letsencrypt/boulder/mail	0.024s
ok  	github.com/letsencrypt/boulder/metrics	0.013s
?   	github.com/letsencrypt/boulder/mocks	[no test files]
ok  	github.com/letsencrypt/boulder/nonce	0.015s
ok  	github.com/letsencrypt/boulder/policy	0.026s
ok  	github.com/letsencrypt/boulder/probs	0.011s
ok  	github.com/letsencrypt/boulder/publisher	5.096s
?   	github.com/letsencrypt/boulder/publisher/proto	[no test files]
	ra_test.go:818: Failed to store certificate: Error 1142: INSERT command denied to user 'sa'@'localhost' for table 'fqdnsets'
--- FAIL: TestTotalCertRateLimit (0.05s)
	ra_test.go:929: [100000] != [99999]
--- FAIL: TestRateLimitLiveReload (2.07s)
FAIL
FAIL	github.com/letsencrypt/boulder/ra	2.986s
ok  	github.com/letsencrypt/boulder/ratelimit	0.022s
fatal error: all goroutines are asleep - deadlock!

goroutine 1 [chan receive]:
testing.RunTests(0x1d09b8, 0x25dc40, 0x6, 0x6, 0x27cf01)
	/usr/local/Cellar/go/1.6.2/libexec/src/testing/testing.go:583 +0x8d2
testing.(*M).Run(0xc82003def8, 0x1991a0)
	/usr/local/Cellar/go/1.6.2/libexec/src/testing/testing.go:515 +0x81
main.main()
	github.com/letsencrypt/boulder/reloader/_test/_testmain.go:64 +0x117

goroutine 11 [chan receive]:
github.com/letsencrypt/boulder/reloader.TestReload(0xc820084240)
	/Users/weppos/go/src/github.com/letsencrypt/boulder/reloader/reloader_test.go:129 +0xc0b
testing.tRunner(0xc820084240, 0x25dca0)
	/usr/local/Cellar/go/1.6.2/libexec/src/testing/testing.go:473 +0x98
created by testing.RunTests
	/usr/local/Cellar/go/1.6.2/libexec/src/testing/testing.go:582 +0x892

goroutine 12 [select]:
github.com/letsencrypt/boulder/reloader.New.func2()
	/Users/weppos/go/src/github.com/letsencrypt/boulder/reloader/reloader.go:51 +0x365
created by github.com/letsencrypt/boulder/reloader.New
	/Users/weppos/go/src/github.com/letsencrypt/boulder/reloader/reloader.go:82 +0x36e
FAIL	github.com/letsencrypt/boulder/reloader	0.031s
ok  	github.com/letsencrypt/boulder/rpc	0.023s
	sa_test.go:378: Couldn't add www.eff.org.der: Error 1142: INSERT command denied to user 'sa'@'localhost' for table 'fqdnsets'
--- FAIL: TestAddCertificate (0.04s)
	sa_test.go:439: Couldn't add test-cert.der: Error 1142: INSERT command denied to user 'sa'@'localhost' for table 'fqdnsets'
--- FAIL: TestCountCertificatesByNames (0.04s)
	sa_test.go:532: Couldn't add www.eff.org.der: Error 1142: INSERT command denied to user 'sa'@'localhost' for table 'fqdnsets'
--- FAIL: TestMarkCertificateRevoked (0.05s)
	sa_test.go:573: Couldn't add www.eff.org.der: Error 1142: INSERT command denied to user 'sa'@'localhost' for table 'fqdnsets'
--- FAIL: TestCountCertificates (0.03s)
	sa_test.go:670: Failed to add name set: Error 1142: INSERT command denied to user 'sa'@'localhost' for table 'fqdnsets'
--- FAIL: TestFQDNSets (0.04s)
	sa_test.go:721: Failed to check FQDN set existence: Error 1142: SELECT command denied to user 'sa'@'localhost' for table 'fqdnsets'
--- FAIL: TestFQDNSetsExists (0.04s)
FAIL
FAIL	github.com/letsencrypt/boulder/sa	0.821s
?   	github.com/letsencrypt/boulder/sa/satest	[no test files]
?   	github.com/letsencrypt/boulder/test	[no test files]
?   	github.com/letsencrypt/boulder/test/ct-test-srv	[no test files]
?   	github.com/letsencrypt/boulder/test/dns-test-srv	[no test files]
ok  	github.com/letsencrypt/boulder/test/mail-test-srv	0.011s
?   	github.com/letsencrypt/boulder/test/vars	[no test files]
ok  	github.com/letsencrypt/boulder/va	15.156s
?   	github.com/letsencrypt/boulder/va/proto	[no test files]
ok  	github.com/letsencrypt/boulder/wfe	0.249s
[!] FAILURE: go test -p 1 github.com/letsencrypt/boulder/akamai github.com/letsencrypt/boulder/bdns github.com/letsencrypt/boulder/ca github.com/letsencrypt/boulder/cdr github.com/letsencrypt/boulder/cmd github.com/letsencrypt/boulder/cmd/admin-revoker github.com/letsencrypt/boulder/cmd/boulder-ca github.com/letsencrypt/boulder/cmd/boulder-publisher github.com/letsencrypt/boulder/cmd/boulder-ra github.com/letsencrypt/boulder/cmd/boulder-sa github.com/letsencrypt/boulder/cmd/boulder-va github.com/letsencrypt/boulder/cmd/boulder-wfe github.com/letsencrypt/boulder/cmd/caa-checker github.com/letsencrypt/boulder/cmd/caa-checker/proto github.com/letsencrypt/boulder/cmd/caa-checker/test-client github.com/letsencrypt/boulder/cmd/cert-checker github.com/letsencrypt/boulder/cmd/contact-exporter github.com/letsencrypt/boulder/cmd/expiration-mailer github.com/letsencrypt/boulder/cmd/expired-authz-purger github.com/letsencrypt/boulder/cmd/notify-mailer github.com/letsencrypt/boulder/cmd/ocsp-responder github.com/letsencrypt/boulder/cmd/ocsp-updater github.com/letsencrypt/boulder/cmd/orphan-finder github.com/letsencrypt/boulder/cmd/rabbitmq-setup github.com/letsencrypt/boulder/cmd/single-ocsp github.com/letsencrypt/boulder/cmd/single-ocsp/test github.com/letsencrypt/boulder/core github.com/letsencrypt/boulder/core/proto github.com/letsencrypt/boulder/csr github.com/letsencrypt/boulder/goodkey github.com/letsencrypt/boulder/grpc github.com/letsencrypt/boulder/grpc/creds github.com/letsencrypt/boulder/log github.com/letsencrypt/boulder/mail github.com/letsencrypt/boulder/metrics github.com/letsencrypt/boulder/mocks github.com/letsencrypt/boulder/nonce github.com/letsencrypt/boulder/policy github.com/letsencrypt/boulder/probs github.com/letsencrypt/boulder/publisher github.com/letsencrypt/boulder/publisher/proto github.com/letsencrypt/boulder/ra github.com/letsencrypt/boulder/ratelimit github.com/letsencrypt/boulder/reloader github.com/letsencrypt/boulder/rpc github.com/letsencrypt/boulder/sa github.com/letsencrypt/boulder/sa/satest github.com/letsencrypt/boulder/test github.com/letsencrypt/boulder/test/ct-test-srv github.com/letsencrypt/boulder/test/dns-test-srv github.com/letsencrypt/boulder/test/mail-test-srv github.com/letsencrypt/boulder/test/vars github.com/letsencrypt/boulder/va github.com/letsencrypt/boulder/va/proto github.com/letsencrypt/boulder/wfe
--------------------------------------------------
---        A unit test or tool failed.         ---
--- Stopping before running integration tests. ---
--------------------------------------------------
```

It is worth to mention I previously installed it months ago but the setup should be fresh, as I cleared the database before restarting. Also, the setup went successfull (no errors).

PS. Congrats, the setup went much nicer than the experience I had the first time. 

/cc @jsha

I provided a patch in #1956. I also noticed two more issues.

The first one is that the following code in `create_db` still caused the permissions to not be properly refreshed.

```
if mysql $dbconn -e 'show databases;' | grep boulder_sa_integration > /dev/null; then
  echo "Databases already created."
  exit 0
fi
```

In fact, it only checks for the presence of the database. But if (like in my case) the database was there but it was an previous installation, then it exited without properly creating the new table and roles. I'm not sure if you prefer to remove the check entirely, or try to make it more clever.

The second issue is that now I get the write permissions correct, but I still have at least one test failing:

```
➜  boulder git:(publicsuffix) ✗ go test ./ra
	ra_test.go:929: [100000] != [99999]
--- FAIL: TestRateLimitLiveReload (2.06s)
FAIL
FAIL	github.com/letsencrypt/boulder/ra	2.901s
```

I'm trying to debug what's going on here.

--- a/test/setup.sh
+++ b/test/setup.sh
@@ -19,6 +19,9 @@ go get \
   github.com/tools/godep \
   golang.org/x/tools/cover &
 
+# Create the database and roles
+./test/create_db.sh &
+
 (curl -sL https://github.com/google/protobuf/releases/download/v2.6.1/protobuf-2.6.1.tar.gz | \
  tar -xzv &&
  cd protobuf-2.6.1 && ./configure --prefix=$HOME && make && make install) &

